hyperspace/core/src/lib.rs

```diff
@@ -50,15 +50,21 @@ where
 {
 	let (mut chain_a_finality, mut chain_b_finality) =
 		(chain_a.finality_notifications().await, chain_b.finality_notifications().await);
+	// `select!` will spawn a new task for each notification, but we need
+	// the notifications to be processed sequentially
+	let lock_a = tokio::sync::Mutex::new(());
+	let lock_b = tokio::sync::Mutex::new(());
 	// loop forever
 	loop {
 		tokio::select! {
 			// new finality event from chain A
 			result = chain_a_finality.next() => {
+				let _guard = lock_a.lock().await;
```

Comment:
why do you need the lock? The loop isn't spawning a new thread.

Reply:
There is a comment [above](https://github.com/ComposableFi/centauri/blob/f90ab1f02ec35b6727b8a718e93427e170980657/hyperspace/core/src/lib.rs#L53)

Reply:
@vmarkushin I went through the docs and code for tokio::select again and it doesn't return until one feature completes, so we don't need the lock

Reply:
The handle of the branch that completed first is executed to completion, the rest are dropped https://github.com/tokio-rs/tokio/blob/80ec80165bfa41c197353983edcbeb64bd45a869/tokio/src/macros/select.rs#L558